lib/api.js

```diff
@@ -0,0 +1,281 @@
+				this.updateLatest( browsers );
+			}
+
+			fn( error, browsers );
+		}.bind( this ) );
+	},
+
+	createWorker: function( options, fn ) {
+		if ( options[ this.versionField ] === "latest" ) {
+			return this.getLatest( options, function( error, version ) {
+				if ( error ) {
+					return fn( error );
+				}
+
+				options = this.extend( {}, options );
+				options[ this.versionField ] = version;
+				this.createWorker( options, fn );
+			}.bind( this ) );
+		}
+
+		var data = JSON.stringify( options );
+		this.request({
+			path: this.path( "/worker" ),
+			method: "POST"
+		}, data, fn );
+	},
+
+	getWorker: function( id, fn ) {
+		this.request({
+			path: this.path( "/worker/" + id )
+		}, fn );
+	},
+
+	changeUrl: function( id, options, fn ) {
+		var data = JSON.stringify( options );
+		this.request({
+			path: this.path( "/worker/" + id + "/url.json" ),
+			method: "PUT"
+		}, data, fn );
+	},
+
+	terminateWorker: function( id, fn ) {
+		this.request({
+			path: this.path( "/worker/" + id ),
+			method: "DELETE"
+		}, fn );
+	},
+
+	getWorkers: function( fn ) {
+		this.request({
+			path: this.path( "/workers" )
+		}, fn );
+	},
+
+	getLatest: function( browser, fn ) {
+		var latest = this.latest,
+		browserId = this.getBrowserId( browser );
+
+		if ( typeof browser === "function" ) {
+			fn = browser;
+			browser = null;
+		}
+
+		// there may be a lot of createWorker() calls with "latest" version
+		// so minimize the number of calls to getBrowsers()
+		if ( this.latestPending ) {
+			return setTimeout(function() {
+				this.getLatest( browser, fn );
+			}.bind( this ), 50 );
+		}
+
+		// only cache browsers for one day
+		if ( !latest || this.latestUpdate < (new Date() - 864e5) ) {
+			this.latestPending = true;
+			return this.getBrowsers(function( error ) {
+				this.latestPending = false;
+
+				if ( error ) {
+					return fn( error );
+				}
+
+				this.getLatest( browser, fn );
+			}.bind( this ) );
+		}
+
+		process.nextTick(function() {
+			fn( null, browser ? latest[ browserId ] : this.extend( {}, latest ) );
+		});
+	},
+
+	takeScreenshot: function( id, fn ) {
+		this.request({
+			path: this.path( "/worker/" + id + "/screenshot.json" )
+		}, fn );
+	},
+
+
+	// internal stuff
+	server: {
+		host: "api.browserstack.com"
+	},
+	latest: null,
+	latestUpdate: 0,
+	latestPending: false,
+
+	path: function( path ) {
+		return "/" + this.version + path;
+	},
+
+	updateLatest: function( browsers ) {
+		var latest = this.latest = {},
+		getBrowserId = this.getBrowserId,
+		versionField = this.versionField;
+
+		this.latestUpdate = new Date();
+		browsers.forEach(function( browser ) {
+			var version = browser[ versionField ],
+			browserId = getBrowserId( browser );
+
+			// ignore devices that don't have versions
+			if ( !version ) {
+				return;
+			}
+
+			// ignore pre-release versions
+			if ( /\s/.test( version ) ) {
+				return;
+			}
+
+			if ( parseFloat( version ) >
+				(parseFloat( latest[ browserId ] ) || 0) ) {
```

Comment:
My IDE automatically reindented everything using tabs, I'll check on the broken `if` line